### PR TITLE
PhpUnitDedicateAssertFixer - remove `is_resource`.

### LINKED ISSUE #6179: Rule `php_unit_dedicate_assert` should not use `assertInternalType` anymore.
Hi,

I just noticed that PHP CS Fixer replace this:

```diff
-        self::assertTrue(is_resource($file));
+        self::assertInternalType('resource', $file);
```

However, the method `TestCase::assertInternalType` does not exist anymore in PHPUnit, [it has been deprecated](https://github.com/sebastianbergmann/phpunit/issues/3369).

On top of that, if just like me you're using the function [`is_resource`](https://www.php.net/manual/en/function.is-resource.php) to check if a resource is open or not, then replacing it with `TestCase::assertInternalType` is wrong because it doesn't check that specific thing. Why? Because `TestCase::assertInternalType` is based on [`getType()`](https://www.php.net/manual/en/function.gettype.php) and does not check the status of the resource.

### PR DESCRIPTION
**Pull request by drupol**
This PR:

- [x] Update `phpunit_dedicate_assert` fixer

Fix #6179

### CODE REVIEW

**Comment by drupol** [comment]
@SpacePossum Here you go, let me know if there is something left to do.

File: tests/Fixer/PhpUnit/PhpUnitDedicateAssertFixerTest.php [hunk 1]
```diff
@@ -307,6 +307,10 @@ public function provideNotFixCases(): \Generator
         yield 'not in class' => [
             '<?php self::assertTrue(is_null($a));',
         ];
+
+        yield 'is_resource' => [
```
Comment (keradus) [inline tests/Fixer/PhpUnit/PhpUnitDedicateAssertFixerTest.php:311]: this PR doesn't provide any reasoning for the change, this test case description also.
please add it here

Comment (drupol) [reply tests/Fixer/PhpUnit/PhpUnitDedicateAssertFixerTest.php]: Added.

File: tests/Fixer/PhpUnit/PhpUnitDedicateAssertFixerTest.php [hunk 2]
```diff
@@ -308,7 +308,10 @@ public function provideNotFixCases(): \Generator
             '<?php self::assertTrue(is_null($a));',
         ];
 
-        yield 'is_resource' => [
+        // Do not replace is_resource() by assertIsResource().
+        // is_resource() also check if the resource is open or closed
```
Comment (keradus) [inline tests/Fixer/PhpUnit/PhpUnitDedicateAssertFixerTest.php:312]:
```suggestion
        // is_resource() also checks if the resource is open or closed,
```

**Comment by SpacePossum** [comment]
Thank you @drupol.

**Pull request closed without merging**